### ISSUE TITLE
Is it possible to recover a restarted master node?

### ISSUE THREAD

**Issue by IvanovOleg** [Azure/acs-engine#2337]
**Is this a request for help?**:
HELP
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of acs-engine?**:
v0.13.0
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes 1.9.3

**What happened**:

Master node VM has been restarted

**What you expected to happen**:

Kubernetes cluster survives restart of the master node

**How to reproduce it** (as minimally and precisely as possible):

restart master vm using Azure portal

**Anything else we need to know**:
```Feb 26 10:47:31 k8s-master-30347309-0 docker[13658]: I0226 10:47:31.845553   13683 kubelet_node_status.go:431] Recording NodeHasSufficientMemory event message for node k8s-master-30347309-0
Feb 26 10:47:31 k8s-master-30347309-0 docker[13658]: I0226 10:47:31.845637   13683 kubelet_node_status.go:431] Recording NodeHasNoDiskPressure event message for node k8s-master-30347309-0
Feb 26 10:47:31 k8s-master-30347309-0 docker[13658]: I0226 10:47:31.845673   13683 kubelet_node_status.go:82] Attempting to register node k8s-master-30347309-0
Feb 26 10:47:31 k8s-master-30347309-0 docker[13658]: E0226 10:47:31.846081   13683 kubelet_node_status.go:106] Unable to register node "k8s-master-30347309-0" with API server: Post https://10.240.0.4:443/api/v1/nodes: dial tcp 10.240.0.4:443: getsockopt
Feb 26 10:47:32 k8s-master-30347309-0 docker[13658]: E0226 10:47:32.819089   13683 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:465: Failed to list *v1.Service: Get https://10.240.0.4:443/api/v1/services?limit=500&resourceVersion=0: dial t
Feb 26 10:47:32 k8s-master-30347309-0 docker[13658]: E0226 10:47:32.828305   13683 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:474: Failed to list *v1.Node: Get https://10.240.0.4:443/api/v1/nodes?fieldSelector=metadata.name%3Dk8s-master-
Feb 26 10:47:32 k8s-master-30347309-0 docker[13658]: E0226 10:47:32.829882   13683 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://10.240.0.4:443/api/v1/pods?fieldSelector=spec.nodeName%3Dk8s-m
Feb 26 10:47:33 k8s-master-30347309-0 docker[13658]: E0226 10:47:33.819805   13683 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:465: Failed to list *v1.Service: Get https://10.240.0.4:443/api/v1/services?limit=500&resourceVersion=0: dial t
Feb 26 10:47:33 k8s-master-30347309-0 docker[13658]: E0226 10:47:33.828853   13683 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:474: Failed to list *v1.Node: Get https://10.240.0.4:443/api/v1/nodes?fieldSelector=metadata.name%3Dk8s-master-
Feb 26 10:47:33 k8s-master-30347309-0 docker[13658]: E0226 10:47:33.830251   13683 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://10.240.0.4:443/api/v1/pods?fieldSelector=spec.nodeName%3Dk8s-m
```

I tried to restart kubelet several times, but without success

**Comment by CecileRobertMichon** [comment]
Hi @IvanovOleg, there is a bug in v0.13.0 where the master nodes don't reboot properly (see #2282). A fix has been merged to master (#2329). In the meantime, you can run `systemctl enable etcd.service` and `systemctl restart etcd` on the master node after reboot.

**Comment by IvanovOleg** [comment]
It works! Thanks!
